diem-move/diem-framework/experimental/sources/MultiTokenBalance.move

```diff
@@ -12,6 +12,19 @@ module DiemFramework::MultiTokenBalance {
         gallery: vector<Token<TokenType>>
     }
 
+    spec TokenBalance {
+        invariant forall i1 in 0..len(gallery), i2 in 0..len(gallery) where gallery[i1].id == gallery[i2].id:
```

Comment:
Because of this invariant, you should be able to use `choose` instead of `choose min`. The use of `choose min` might be confusing for the user.

Also wonder in general whether it makes sense to introduce a helper function `find_token_by_id` or `find_token_index_by_id`, as this pattern appears frequently in your spec.